go/private/mode.bzl

```diff
@@ -210,6 +224,9 @@ def link_mode_args(mode):
     elif mode.link == LINKMODE_PLUGIN:
         if platform in _LINK_PLUGIN_PLATFORMS:
             args.append("-dynlink")
+    elif mode.link == LINKMODE_PIE:
+        if platform in _LINK_PIE_PLATFORMS:
+          args.append("-shared")
```

Comment:
I don't think this is enough. See src/cmd/go/internal/work/init.go for how `-buildmode="pie"` is interpreted. It passes `-shared` to the compiler. I think it also passes `-buildmode=pie` to the linker. And `-pie` is passed to `cgo -dynimport`.

Also, this indentation doesn't look correct. Make sure to run buildifier to format this file.

Reply:
Done.
Code now passes pie to cgo action. -buildmode=pie is already being passed in link.bzl whenever linkmode != LINKMODE_NORMAL.

Reply:
Added test. Basing check for PIE on https://android.googlesource.com/platform/bionic/+/master/linker/linker_main.cpp#388. PIE executables are implemented as shared libraries (type ET_DYN). Other than that, I think any kind of relocation in the text section is a no-no.